### PR TITLE
Holiday Creator Features Rewrite

### PR DESCRIPTION
**Pull request by SmokeyStack**
This PR aims to resolve: https://github.com/Bedrock-OSS/bedrock-wiki/projects/3

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -1,16 +1,20 @@
 {
-	"trailingComma": "es5",
-	"useTabs": true,
-	"tabWidth": 4,
+	"embeddedLanguageFormatting": "off",
+	"printWidth": 100,
 	"semi": false,
 	"singleQuote": true,
-	"printWidth": 80,
```
Comment (ThomasOrs) [inline .prettierrc.json:7]: Change from default print width?

File: docs/blocks/block-tags.md
Comment (ThomasOrs) [inline docs/blocks/block-tags.md:1]: The Experiemntal warning for the `q.block` molang queries is out of date. These established in 1.20.60, the toggle was also removed. Might be worth grabbing as part of this pages changes.

File: docs/blocks/block-states.md
Comment (ThomasOrs) [inline docs/blocks/block-states.md:1]: File still contains `Event Response
` with mention of turning on HCF.

File: docs/blocks/block-models.md [hunk 2]
```diff
@@ -94,50 +83,58 @@ The "paper_bag" model has multiple pre-made textures, listed below:
 
 -   `textures/blocks/paper_bag.png`
 
-    <WikiImage src="/assets/images/blocks/block-models/paper_bag.png" style="background-color: rgb(0,0,0,0.15);" pixelated="true" width="128"/>
-    <br>
-    <br>
+            <WikiImage src="/assets/images/blocks/block-models/paper_bag.png" style="background-color: rgb(0,0,0,0.15);" pixelated="true" width="128"/>
```
Comment (ThomasOrs) [inline docs/blocks/block-models.md]: This imagine and the <br> tag are displaying as part of the code highlight. 
image

File: docs/blocks/block-models.md [hunk 3]
```diff
@@ -94,50 +83,58 @@ The "paper_bag" model has multiple pre-made textures, listed below:
...
+        <br>
+
+    <br>
 
 -   `textures/blocks/paper_bag_bottom_fold.png`
 
-    <WikiImage src="/assets/images/blocks/block-models/paper_bag_bottom_fold.png" style="background-color: rgb(0,0,0,0.15);" pixelated="true" width="128"/>
-    <br>
-    <br>
+            <WikiImage src="/assets/images/blocks/block-models/paper_bag_bottom_fold.png" style="background-color: rgb(0,0,0,0.15);" pixelated="true" width="128"/>
```
Comment (ThomasOrs) [inline docs/blocks/block-models.md]: See above comment

File: docs/blocks/precise-rotation.md
Comment (ThomasOrs) [inline docs/blocks/precise-rotation.md:1]: Page still has a requires HCF warning

File: docs/blocks/rotatable-blocks.md
Comment (ThomasOrs) [inline docs/blocks/rotatable-blocks.md:1]: Page contains HCF example?
Log rotation example with Unit Cube

File: docs/blocks/custom-fluids.md
Comment (ThomasOrs) [inline docs/blocks/custom-fluids.md:1]: Contains a warning for requires HCF for Molang queries - Not needed
Contains queued_ticking which no longer exists
Should this be hidden?

File: docs/items/item-components.md [hunk 4]
```diff
@@ -54,23 +55,35 @@ Determines whether an item can be placed in the off-hand slot of the inventory.
 
 ## Block Placer
 
-Sets the item as a Planter item component for blocks. Items with this component will place a block when used.
+Block Placer item component. Items with this component will place a block when used. Experimental toggles required: Holiday Creator Features (in format versions before 1.20.10).
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: Should experimental toggles required be included here? The toggle no longer exists 🤔

File: docs/items/item-components.md [hunk 5]
```diff
@@ -82,20 +95,55 @@ Determines if an item will break blocks in Creative Mode while swinging.
 
 ## Cooldown
 
-Sets an items "Cool down" time. After using an item, it becomes unusable for the duration specified by the 'duration' setting of this component.
+Cool down time for a component. After use, all items in a specified 'cool down category' become unusable for a determined amount of time defined in the component. In format versions before 1.20.10, this component requires the 'Holiday Creator Features' experimental toggle.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 6]
```diff
@@ -127,14 +188,18 @@ Allows a creator to determine how quickly an item can dig specific blocks.
 
 ## Display Name
 
-Sets the item display name within Minecraft: Bedrock Edition. This component may also be used to pull from the localization file by referencing a key from it.
+Defines the text shown when an item name is shown, such as hover text. In format versions prior to 1.20.0, this component requires the 'Holiday Creator Features' experimental toggle.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 7]
```diff
@@ -144,69 +209,100 @@ Sets the item display name within Minecraft: Bedrock Edition. This component may
...
-"minecraft:display_name":{
+"minecraft:display_name": {
     "value": "item.snowball.name"
 }
 ```
 
 ## Durability
 
-Sets how much damage the item can take before breaking, and allows the item to be combined at an anvil, grindstone, or crafting table.
+Determines how much damage the item can take before breaking, and allows the item to be combined in crafting. In format versions prior to 1.20.0, this component requires the 'Holiday Creator Features' experimental toggle.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 8]
```diff
@@ -220,48 +316,89 @@ Determines the item's enchantability, influencing the quality and quantity of po
...
+"minecraft:enchantable": {
+    "slot": "sword",
+    "value": 10
+}
+```
 
 ## Entity Placer
 
-Allows an item to place entities into the world. Additionally, in version 1.19.80 and above, the component allows the item to set the spawn type of a monster spawner.
+Allows the item to place specified entities into the world. In format versions prior to 1.20.0, this component requires the 'Holiday Creator Features' experimental toggle.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 9]
```diff
@@ -220,48 +316,89 @@ Determines the item's enchantability, influencing the quality and quantity of po
...
+    "nutrition": 3,
+    "saturation_modifier": 0.6,
     "using_converts_to": "bowl"
 }
 ```
 
 ## Fuel
 
-Allows this item to be used as fuel in a furnace to 'cook' other items.
+Allows the item to be used as fuel in a furnace to 'cook' other items. In format versions prior to 1.20.0, this component requires the 'Holiday Creator Features' experimental toggle.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 10]
```diff
@@ -292,38 +435,67 @@ Determines if an item is rendered like a tool while in-hand.
...
 
 ```json
-"minecraft:hover_text_color": "green"
+"minecraft:hover_text_color": "minecoin_gold"
 ```
 
 ## Icon
 
-Sets the icon item component. Determines the icon to represent the item in the UI and elsewhere.
+Determines the icon to represent the item in the UI and elsewhere. In format versions before 1.20.10, this component requires the 'Holiday Creator Features' experimental toggle.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 11]
```diff
@@ -346,20 +520,40 @@ Determines how many of an item can be stacked together.
 
 ## Projectile
 
-Compels the item to shoot, similarly to an arrow. Items with `minecraft:projectile` can be shot from dispensers or used as ammunition for items with the `minecraft:shooter` item component. Additionally, this component sets the entity that is spawned for items that also contain the `minecraft:throwable` component.
+Projectile item component. Projectile items shoot out, like an arrow. In format versions prior to 1.20.10, this component requires the 'Holiday Creator Features' experimental toggle.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 12]
```diff
@@ -377,24 +571,87 @@ Listed [here](https://learn.microsoft.com/en-us/minecraft/creator/reference/cont
 
 ## Repairable
 
-Defines the items that can be used to repair a defined item, and the amount of durability each item restores upon repair. Each entry needs to define a list of strings for 'items' that can be used for the repair and an optional 'repair_amount' for how much durability is repaired.
+Repairable item component: Determines which items can be used to repair a defined item, as well as the amount of durability specified items will repair. In format versions prior to 1.20.10, this component requires the 'Holiday Creator Features' experimental toggle.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 13]
```diff
@@ -377,24 +571,87 @@ Listed [here](https://learn.microsoft.com/en-us/minecraft/creator/reference/cont
...
+            "repair_amount": "math.min(q.remaining_durability + c.other->q.remaining_durability + math.floor(q.max_durability /20), c.other->q.max_durability)" // Vanilla formula
+        }
+    ]
 }
 ```
 
 ## Shooter
 
-Compels an item to shoot projectiles, similarly to a bow or crossbow. Must have the `minecraft:use_modifiers` component in order to function properly.
+Shooter Item Component. Must have the `minecraft:use_modifiers` component in order to function properly. In format versions prior to 1.20.10, this component requires the 'Holiday Creator Features' experimental toggle.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 14]
```diff
@@ -454,70 +718,114 @@ Determines which tags are included on a given item.
 
 ## Throwable
 
-Sets the throwable item component.
+Throwable item component. Throwable items, such as a snowball. In format versions prior to 1.20.10, this component requires the 'Holiday Creator Features' experimental toggle. Item must have the `minecraft:projectile` component.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above

File: docs/items/item-components.md [hunk 15]
```diff
@@ -454,70 +718,114 @@ Determines which tags are included on a given item.
...
-    "movement_modifier": 0.35
+    "movement_modifier": 0.5,
+    "use_duration": 1.0
 }
 ```
 
 ## Wearable
 
-Sets the wearable item component.
+Determines where the item can be worn. If any non-hand slot is chosen, the max stack size is set to 1. Experimental toggles required: Holiday Creator Features.
```
Comment (ThomasOrs) [inline docs/items/item-components.md]: See above / this one does not say what version it was introduced at all.

File: docs/items/troubleshooting-items.md
Comment (ThomasOrs) [inline docs/items/troubleshooting-items.md:1]: One of the tip implies all items 1.16.100 are experimental, which is inaccurate.

File: docs/items/custom-armor.md [hunk 16]
```diff
@@ -35,66 +32,68 @@ Create a chest piece:
...
-				"dispensable": true,
-				"slot": "slot.armor.chest"
-			},
-			// Provide its durability
-			"minecraft:durability": {
-				"max_durability": 200
-			}
-		}
-	}
+    "format_version": "1.16.100",
```
Comment (ThomasOrs) [inline docs/items/custom-armor.md]: Still has an old-format version

File: docs/items/custom-armor.md [hunk 17]
```diff
@@ -163,93 +166,95 @@ So while the chest piece alone is great, you probably want a whole set, so from
...
-			"minecraft:wearable": {
-				"dispensable": true,
-				"slot": "slot.armor.legs"
-			},
-			"minecraft:durability": {
-				"max_durability": 200
-			}
-		}
-	}
+    "format_version": "1.16.100",
```
Comment (ThomasOrs) [inline docs/items/custom-armor.md]: Old format version

File: docs/items/custom-armor.md [hunk 18]
```diff
@@ -263,91 +268,93 @@ This is just like the chest piece, just we change some of the categories and slo
...
-			"minecraft:wearable": {
-				"dispensable": true,
-				"slot": "slot.armor.head"
-			},
-			"minecraft:durability": {
-				"max_durability": 200
-			}
-		}
-	}
+    "format_version": "1.16.100",
```
Comment (ThomasOrs) [inline docs/items/custom-armor.md]: old format version

File: docs/items/custom-armor.md [hunk 19]
```diff
@@ -361,91 +368,93 @@ You already know the pattern so lets make the item and attachable json files.
...
-			"minecraft:wearable": {
-				"dispensable": true,
-				"slot": "slot.armor.feet"
-			},
-			"minecraft:durability": {
-				"max_durability": 200
-			}
-		}
-	}
+    "format_version": "1.16.100",
```
Comment (ThomasOrs) [inline docs/items/custom-armor.md]: Old format version

File: docs/items/equipped-item-commands.md
Comment (ThomasOrs) [inline docs/items/equipped-item-commands.md:1]: Page contains mention of requiring HCF

File: docs/items/tool-durability.md
Comment (ThomasOrs) [inline docs/items/tool-durability.md:1]: It doesn't seem clear on this page that the Item Event method no longer works.

Comment (ThomasOrs) [inline docs/items/tool-durability.md:1]: There is a warning that this uses experimental scripts `1.9.0-beta` are the apis now stable?

**Review by ThomasOrs — CHANGES_REQUESTED** [review]
Looks good overall! Some cases where some HCF examples / warnings still appear.
In some places it is not clear that post 1.21.20 there HCF examples / methods 100% will not work.

File: docs/items/item-components.md (hunk 4 above)
Comment (ThomasOrs) [reply docs/items/item-components.md]: Should this instead be ` Requires format version 1.20.10 or above`?

**Review by SmokeyStack — COMMENTED** [review]
Applied suggestions

**Comment by ThomasOrs** [comment]
:shipit:

**Pull request merged**